presto-docs/src/main/sphinx/functions/window.rst

```diff
@@ -5,29 +5,67 @@ Window Functions
 Window functions perform calculations across rows of the query result.
 They run after the ``HAVING`` clause but before the ``ORDER BY`` clause.
 Invoking a window function requires special syntax using the ``OVER``
-clause to specify the window. A window has three components:
+clause to specify the window as follows::
 
-* The partition specification, which separates the input rows into different
-  partitions. This is analogous to how the ``GROUP BY`` clause separates rows
-  into different groups for aggregate functions.
-* The ordering specification, which determines the order in which input rows
-  will be processed by the window function.
-* The window frame, which specifies a sliding window of rows to be processed
-  by the function for a given row. If the frame is not specified, it defaults
-  to ``RANGE UNBOUNDED PRECEDING``, which is the same as
-  ``RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW``. This frame contains all
-  rows from the start of the partition up to the last peer of the current row.
+    function(args) OVER (
+        [PARTITION BY expression]
+        [ORDER BY expression [ASC|DESC]]
+        [frame]
+    )
+
+A ``frame`` is one of::
+
+    {RANGE|ROWS} frame_start
+    {RANGE|ROWS} BETWEEN frame_start AND frame_end
+
+``frame_start`` and ``frame_end`` can be any of::
+
+    UNBOUNDED PRECEDING
+    expression PRECEDING  -- only allowed in ROWS mode
+    CURRENT ROW
+    expression FOLLOWING  -- only allowed in ROWS mode
+    UNBOUNDED FOLLOWING
 
-For example, the following query ranks orders for each clerk by price::
+
+The window definition has 3 components:
+
+* The ``PARTITION BY`` clause separates the input rows into different
+  partitions. This is analogous to how the ``GROUP BY`` clause separates rows
+  into different groups for aggregate functions.  If ``PARTITION BY`` is not
+  specified, the entire input is treated as a single partition.
+* The ``ORDER BY`` clause determines the order in which input rows
+  will be processed by the window function.  If ``ORDER BY`` is not specified,
+  the ordering is undefined.
+  **Note that the ORDER BY clause within window functions does not support ordinals. You need to use actual expressions**
+* The ``frame`` clause specifies the sliding window of rows to be processed by the
```

Comment:
This is much better explained than before! I would say the only thing missing is describing that the RANGE mode focuses on selecting a frame based on the value of the row?

And this doesn't apply for now, as anything other than UNBOUNDED is not implemented, but would <i> PRECEDING mean that we take the value at CURRENT ROW, do - <i> and that would be the start value? Or would the start value be the value from row index - <i>

Reply:
Decided to leave out too detailed a description of range since it's not relevant for now.  But you are correct that (int) preceding would be value at CURRENT ROW - (int) for the start value.